### ISSUE TITLE
➕ Feature - Export DIFFERENCE Type so it can be imported in Ts projects

### ISSUE THREAD

**Issue by GabrielCTroia** [AsyncBanana/microdiff#37]
**❗ Is your feature request related to a problem? Please describe.**
Yes. I cannot use the ReturnType<typeof microdiff> in Typescript because it fails with this error:

Exported variable 'myFn' has or is using name 'DifferenceCreate' from external module "node_modules/microdiff/dist/index" but cannot be named.ts(4023)

**✅ Describe the solution you'd like**
I believe simply exporting all the types would be sufficient, or putting them under the "microdiff" module.

**↗️ Describe alternatives you've considered**
None yet

**➕ Additional context**
This happens with Typescript 4.9

Cheers!

**Comment by senseibaka** [comment]
Hey @GabrielCTroia this is a workaround, but typescript does let you extract the return type of a function using `ReturnType`, like so:

```typescript
import diff from 'microdiff';

export type Difference = ReturnType<typeof diff>[0];
```

**Comment by AsyncBanana** [comment]
This should be fixed in v1.3.2
